code/modules/hallucinations/effects/major.dm

```diff
@@ -408,3 +408,66 @@
 	if(images && images[1])
 		images[1].icon = 'icons/mob/alien.dmi'
 		images[1].icon_state = "alienh_pounce"
+
+/**
+  * # Hallucination - Sniper
+  *
+  * Fires a penetrator round at the target. On hit, knockdown + stam loss + hallucinated blood splatter for a bit.
+  */
+ /obj/effect/hallucination/sniper
+	duration = 15 SECONDS
+
+/obj/effect/hallucination/sniper/Initialize(mapload, mob/living/carbon/target)
+	. = ..()
+
+	// Find a start spot for the sniper bullet
+	var/list/possible_turfs = list()
+	for(var/turf/t in range(world.view + 5, target))
+		if(get_dist(t, target.loc) > world.view)
+			possible_turfs += t
+	if(!length(possible_turfs))
+		return
+
+	// Fire the bullet
+	var/turf/shot_loc = get_turf(pick(possible_turfs))
+	var/obj/item/projectile/bullet/sniper/penetrator/hallucination/bullet = new(shot_loc)
```

Comment:
this means that everyone will be able to see it, no?

Reply:
Yep, you need to make a hallucination that will act like a bullet - or make this one invisible and only show it to the hallucinator

Reply:
In order to avoid redefining the entirety of a projectile and how firing is operated, I'll take a normal bullet, make it invisible, then add an image overlay to it so only the hallucinator can see it.